### ISSUE TITLE
Feature Request: Add a `preduce_sum` method

### ISSUE THREAD

**Issue by chaserileyroberts** [jax-ml/jax#839]
Currently, the way to do a "reduce sum" while running on distributed hardware is to do `psum`. However, `psum` copies the resulting sum to all workers, which can be wasteful if you only want 1 copy. I would like a `preduce_sum` that takes the sum, but only writes it to a single worker, thus avoiding the extra communication overhead. (This would also in turn totally remove the `pmap`ed axis, so you would have to run it outside of a `pmap`).

**Comment by froystig** [comment]
I'll close this due to inactivity over time. These days we have [`shard_map`](https://jax.readthedocs.io/en/latest/notebooks/shard_map.html) for manual parallelism, so hopefully the operation in mind here could be implemented with a shmap.
